Mathlib/SetTheory/Cardinal/Finite.lean

```diff
@@ -149,4 +149,123 @@ theorem card_eq_top_of_infinite [Infinite α] : card α = ⊤ :=
   mk_toPartENat_of_infinite
 #align part_enat.card_eq_top_of_infinite PartENat.card_eq_top_of_infinite
 
+theorem card_congr {α : Type _} {β : Type _} (f : α ≃ β) : PartENat.card α = PartENat.card β :=
+  Cardinal.toPartENat_congr f
+#align part_enat.card_congr PartENat.card_congr
+
+theorem card_uLift (α : Type _) : card (ULift α) = card α :=
+  card_congr Equiv.ulift
+#align part_enat.card_ulift PartENat.card_uLift
+
+@[simp]
+theorem card_pLift (α : Type _) : card (PLift α) = card α :=
+  card_congr Equiv.plift
+#align part_enat.card_plift PartENat.card_pLift
+
+theorem card_image_of_injOn {α : Type u} {β : Type v} {f : α → β} {s : Set α} (h : Set.InjOn f s) :
+    card (f '' s) = card s :=
+  card_congr (Equiv.Set.imageOfInjOn f s h).symm
+#align part_enat.card_image_of_inj_on PartENat.card_image_of_injOn
+
+theorem card_image_of_injective {α : Type u} {β : Type v} (f : α → β) (s : Set α)
+    (h : Function.Injective f) : card (f '' s) = card s :=
+  card_congr (Equiv.Set.imageOfInjOn f s (Set.injOn_of_injective h s)).symm
+#align part_enat.card_image_of_injective PartENat.card_image_of_injective
+
+-- Add other similar lemmas?
+theorem succ_le_iff {n : ℕ} {e : PartENat} : ↑n.succ ≤ e ↔ ↑n < e := by
+  rw [coe_lt_iff, coe_le_iff]
+  apply forall_congr'; intro a; rw [Nat.succ_le_iff]
+#align part_enat.succ_le_iff PartENat.succ_le_iff
+
+/- The line
+-- obtain ⟨m, hm⟩ := cardinal.lt_aleph_0.mp h,
+extract an integer m from a cardinal c such that h : c < ℵ₀
+It may appear easier to use than the rewrites I finally use … -/
+theorem natCast_le_iff_le {n : ℕ} {c : Cardinal} : ↑n ≤ toPartENat c ↔ ↑n ≤ c := by
```

Comment:
This should move to `Cardinal/Basic` (and also the following lemmas). I also think it would be a lot nicer to state this as 
```lean
theorem toPartENat_le_iff_of_le_aleph0 {c c' : Cardinal} (h : c ≤ ℵ₀) : 
  toPartENat c ≤ toPartENat c' ↔ c ≤ c' := sorry
```
(you could also have a version that has hypothesis `c' ≤ ℵ₀`). Then your lemma (if you still need it) follows from this lemma and `toPartENat_cast`.

Reply:
I think you shouldn't keep your version of this (and the following) lemmas, unless you are using them again.
If you do keep them, you should rename them to something like `natCast_le_toPartENat_iff`.

Reply:
I did this lemmas, and two other : `toPartENat.mono` and `toPartENat_eq_top_iff`

Reply:
(The version with `c' \le \aleph_0` is false.)